File: api/src/routes/consultations.ts
import { Hono } from 'hono';
import { Env } from '../types';
import { requireAuth } from '../auth';

const router = new Hono<{ Bindings: Env }>();

router.post('/', async (c) => {
  const user = await requireAuth(c.req.header('authorization'), c.env, c.env.DB);
  const { title, description, practice_area, mode, scheduled_date, duration_minutes } = await c.req.json();
  const { meta } = await c.env.DB.prepare(
    'INSERT INTO consultations (client_id, title, description, practice_area, mode, scheduled_date, duration_minutes, fee, status) VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?)'
  ).bind(user.id, title, description || null, practice_area || null, mode || 'video', scheduled_date || null, duration_minutes || 30, 1500, 'scheduled').run();
  const cons = await c.env.DB.prepare('SELECT id, title, status FROM consultations WHERE id = ?').bind(meta.last_row_id).first();
  return c.json({ message: 'Consultation booked', consultation: cons });
});

router.get('/', async (c) => {
  const user = await requireAuth(c.req.header('authorization'), c.env, c.env.DB);
  let query = 'SELECT id, title, practice_area, mode, status, scheduled_date, fee, created_at FROM consultations WHERE client_id = ? ORDER BY created_at DESC';
  if (user.role === 'lawyer') query = 'SELECT id, title, practice_area, mode, status, scheduled_date, fee, created_at FROM consultations WHERE lawyer_id = ? ORDER BY created_at DESC';
  const result = await c.env.DB.prepare(query).bind(user.id).all();
  return c.json(result.results);
});

router.get('/:id', async (c) => {
  const cons = await c.env.DB.prepare('SELECT * FROM consultations WHERE id = ?').bind(c.req.param('id')).first();
  if (!cons) return c.json({ detail: 'Consultation not found' }, 404);
  return c.json(cons);
});

router.patch('/:id/status', async (c) => {
  const status = c.req.query('status');
  await c.env.DB.prepare('UPDATE consultations SET status = ? WHERE id = ?').bind(status, c.req.param('id')).run();
  return c.json({ message: `Consultation status updated to ${status}` });
});

export default router;

File: backend/app/routes/consultations.py
from fastapi import APIRouter, Depends, HTTPException
from pydantic import BaseModel
from sqlalchemy.orm import Session
from datetime import datetime
from app.database import get_db
from app.models.consultation import Consultation, ConsultationStatus, ConsultationMode
from app.models.user import User, UserRole
from app.utils.auth import get_current_user

router = APIRouter(prefix="/api/consultations", tags=["Consultations"])

class ConsultationRequest(BaseModel):
    title: str
    description: str | None = None
    practice_area: str | None = None
    mode: str = "video"
    scheduled_date: str | None = None
    duration_minutes: int = 30

@router.post("/")
def create_consultation(req: ConsultationRequest, current_user: User = Depends(get_current_user), db: Session = Depends(get_db)):
    consultation = Consultation(
        client_id=current_user.id,
        title=req.title,
        description=req.description,
        practice_area=req.practice_area,
        mode=req.mode,
        scheduled_date=datetime.fromisoformat(req.scheduled_date) if req.scheduled_date else None,
        duration_minutes=req.duration_minutes,
        fee=1500,
        status=ConsultationStatus.SCHEDULED,
    )
    db.add(consultation)
    db.commit()
    db.refresh(consultation)
    return {"message": "Consultation booked", "consultation": {"id": consultation.id, "title": consultation.title, "status": consultation.status.value}}

@router.get("/")
def list_consultations(current_user: User = Depends(get_current_user), db: Session = Depends(get_db)):
    query = db.query(Consultation)
    if current_user.role == UserRole.CLIENT:
        query = query.filter(Consultation.client_id == current_user.id)
    elif current_user.role == UserRole.LAWYER:
        query = query.filter(Consultation.lawyer_id == current_user.id)
    consultations = query.all()
    return [
        {
            "id": c.id,
            "title": c.title,
            "practice_area": c.practice_area,
            "mode": c.mode.value if c.mode else None,
            "status": c.status.value if c.status else None,
            "scheduled_date": str(c.scheduled_date) if c.scheduled_date else None,
            "fee": c.fee,
            "created_at": str(c.created_at),
        }
        for c in consultations
    ]

@router.get("/{consultation_id}")
def get_consultation(consultation_id: int, current_user: User = Depends(get_current_user), db: Session = Depends(get_db)):
    consultation = db.query(Consultation).filter(Consultation.id == consultation_id).first()
    if not consultation:
        raise HTTPException(status_code=404, detail="Consultation not found")
    return {
        "id": consultation.id,
        "title": consultation.title,
        "description": consultation.description,
        "practice_area": consultation.practice_area,
        "mode": consultation.mode.value if consultation.mode else None,
        "status": consultation.status.value,
        "scheduled_date": str(consultation.scheduled_date) if consultation.scheduled_date else None,
        "duration_minutes": consultation.duration_minutes,
        "fee": consultation.fee,
        "notes": consultation.notes,
        "created_at": str(consultation.created_at),
    }

@router.patch("/{consultation_id}/status")
def update_consultation_status(consultation_id: int, status: str, current_user: User = Depends(get_current_user), db: Session = Depends(get_db)):
    consultation = db.query(Consultation).filter(Consultation.id == consultation_id).first()
    if not consultation:
        raise HTTPException(status_code=404, detail="Consultation not found")
    consultation.status = ConsultationStatus(status)
    db.commit()
    return {"message": f"Consultation status updated to {status}"}
